ocaml/xcp-rrdd/bin/rrdd/rrdd_server.ml

```diff
@@ -456,6 +457,30 @@ let query_host_ds (ds_name : string) : float =
           query ds_name rrdi
   )
 
+(** Dump all latest data of host dss to file in json format so that any client
+    can read even if it's non-privileged user, such as NRPE. *)
+let dump_host_dss_to_file (file : string) : unit =
+  let json =
+    with_lock mutex (fun () ->
+        match !host_rrd with
+        | None ->
+            `Assoc []
+        | Some rrdi ->
+            `Assoc
+              (Rrd.ds_names rrdi.rrd
```

Comment:
I wrote test code to dump `stats` in `do_monitor_write`, but found some data is only the raw data than calculated as Data Source Type.
e.g. below is CPU time
```
[root@xrtmia-11-07 ~]# json_reformat< /dev/shm/metrics/host-dss1 | grep cpu_avg
    "cpu_avg": 34199.011874511874,
```
while the queried data from rrd is correct CPU use percentage
```
[root@xrtmia-11-07 ~]# json_reformat< /dev/shm/metrics/host-dss | grep cpu_avg
    "cpu_avg": 0.5052633881568909,
```
So looks still need to query from rrd even if there are about 100~200 metrics query every 5 seconds.

Reply:
Or this thing that collects the data (e.g. Nagios or its plugin) processes the raw data directly?

Reply:
Ah, that makes sense, the derivation step on the data needs to happen in some sort of way

Reply:
Nagios plugin can processes the raw data indeed, while that's duplicated with rrd, and it has to query 2 times with 5 seconds  interval that will slow down Nagios check obviously. 
If you concern the query count very much, I think we can add `filter` to let only specified `ds_name`(s) data are queried. So in future any people want to dump some data they should update the filter firstly.

Reply:
I don't know about twice in a 5 sec interval. It just needs to take regular samples at some interval and calculate whatever it needs? Doesn't Nagios actually _want_ the raw data, before averages/min/max etc are taken, so that it can do that based on it's own time scales?

Reply:
Nagios NRPE won't collect and save data (like rrd), it just handle a check calling from client and execute corresponding check command, e.g.  `command[check_cpu]=/usr/lib64/nagios/plugins/check_cpu -w 80% -c 90%` , `check_cpu` internally run dom0 cmd `mpstat` to get current cpu usage, and finally return to client.
This process is similar to user executes OS command through SSH.
If NRPE reads the raw data from rrd dump file, then `check_cpu` has to read the file, sleep 5s, read again, calculate the result by self. This duplicates with rrd, and more bad thing is the client will have to wait 5s to get check result.

Reply:
I agree it's not worth duplicating the calculation inside dom0, so I think it's fine as it is.

As a note, for prometheus / openmetrics the values are exported as raw values, the appliance gathering metrics deal with these

Reply:
OK that's fine then.